packages/3id-did-resolver/src/__tests__/vectors.json

```diff
@@ -159,13 +159,13 @@
       }
     },
     "http://localhost:7007/api/v0/documents": [{
-      "expectedBody": {"doctype":"tile","genesis":{"header":{"controllers":["did:key:zQ3shcMwX6vEZkbCWrZqL5W47yditTzauxC5x4WNbgo5r2PbW"],"family":"3id"}},"docOpts":{"anchor":false,"publish":false}},
+      "expectedBody": {"doctype":"tile","genesis":{"header":{"controllers":["did:key:zQ3shcMwX6vEZkbCWrZqL5W47yditTzauxC5x4WNbgo5r2PbW"],"family":"3id"}},"docOpts":{"anchor":false,"publish":false,"sync":true}},
```

Comment:
Defaults now happen at the highest up function that is the entry point from the user's application code into the ceramic system.  In this case that is the http client.  So the http client fills in all the defaults client-side, and makes the HTTP request to the daemon with a fully-populated `opts` object, with defaults already filled in.  This is good, because if, for example, we ever changed any of the defaults, the behavior would change only when the user upgrades their application to the new client, not when the server gets upgraded, which may be done independently.

Reply:
Again, I am extremely worried about breaking the http api at this point since if we don't have time to release the updated APIs before ELP starts we will run into issues.

Reply:
This isn't breaking the http api.  The client is just sending `opts` explicitly where it used not to.  Old daemons should still be able to process these requests

Reply:
Ok, so sync will default to true even if not passed?

Reply:
Yep!

Reply:
`documents` endpoint hits `CeramicDaemon.createFromGenesis`, which calls `Ceramic.createDocumentFromGenesis`, which [sets defaults](https://github.com/ceramicnetwork/js-ceramic/blob/df6ee247957af9fd3f29c4a9b70e1ce57bf90cf9/packages/core/src/ceramic.ts#L498)

Reply:
ok, ty. All good then :+1: